### DISCUSSION TITLE
Bluetooth speaker connection plays some albums but not others

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#12] [Fixed Issues] [answered]
[raudio-1_1.docx](https://github.com/rern/rAudio-1/files/5725240/raudio-1_1.docx)
[raudio-1_1.zip](https://github.com/rern/rAudio-1/files/5725247/raudio-1_1.zip)

Rern (please forgive the duplication with previous PM's)

PI Zero W
rAudio-1 (20201219 and updated to 202011221-1)
HK Soundsticks  Bluetooth enabled speakers
No other sound card installed on PI Zero W

Re-flashing the SD card with this mornings image (20201219) seems to have solved my Bluetooth connection issues with this setup. I now have sound output (mostly). 

I experiencing a very weird problem in that some albums will play and others will not. I have narrowed it down to those albums which show 'f-bit' in the playback details will not play but those which display '16 bit' in the playback details will play. The uploaded file shows the problem in more detail.

A separate PI3B+ running the same rAudio-1 version and updates, but with an IQAudIO DAC+, has no problem playing any of these albums.

Chris

**Comment by EarsofTin** [comment]
I ran the 20201221 update this morning and BT is now working in both directions (sender and sink).  Trying different files and have not encountered a problem file so far.

Edited to add playing mp3s of various bit rates and FLAC from CD rips without problem but sometimes stuttering if speaker is connected midstream.  Pair first then start playback is okay.

**Comment by ceejayemm** [comment]
Update:

It seems that the albums that will not play are recording in 'Stereo, 44100Hz and 32 bit float' whilst those tracks that will play are recorded in 'Stereo, 44100Hz and 16 bit PCM' (according to Audacity v2.2.2). 

This means nothing to me, is there a means of converting the former to the latter? Presumable the DAC in my other installations is taking care of the conversion ? Can rAudio be enabled to do the conversion on the fly when using BT speakers and no other sound card ? 

Chris

**Reply by ceejayemm** [reply]
Further Update

I have found that I can use Audacity to convert tracks to 16 bit PCM and then they will play via the BT speakers. Its a bit laborious and I wouldn't want to do it too often so if a means of playing '32 bit float' encoded files in my scenario can be found that would be great.

Thanks

Chris

**Reply by cYCL157** [reply]
out of curiosity -- do you have ffmpeg enabled on the mpd page?  i understood that to help with codec conversions

**Reply by ceejayemm** [reply]
Yes, FFMpeg enabled. I have also fiddled with the SOXR settings ( but put them back to default as it didn't seem to have any effect - and I don't know what SOXR does anyway.

Chris

**Comment by ceejayemm** [comment]
Any further thoughts on this niggling little problem - anybody ?

Chris

**Comment by ceejayemm** [comment]
@rern 

It appears that (most) Bluetooth devices can only play audio recorded up to 24 bit hence why my albums recorded at 32 bit floating point won't play. Most of these albums are downloads from iTunes or Amazon so this is not something I have done out of choice.  It seems that there might be a means of getting MPD to play these files, see:

https://wiki.archlinux.org/index.php/Music_Player_Daemon/Tips_and_tricks#Example_configuration:_Output_with_44.1_KHz_at_e._g._16_bit_depth,_multiple_programs_at_once

I am not bothered about being able to play to multiple programs as in this example. I have tried to implement this by making changes to /etc/mpd.conf manually for the BT Soundsticks I use only but RAudio seems to overwrite this file on occasion so I haven't been able to get it to work.

I realise that this might compromise the efforts to produce the the best sound quality possible and that mine might be a unique use case as nobody else has responded to this thread but if it gets more music playable in this case it might be worthwhile. Would you consider making these changes to the RAudio configuration where Bluetooth is the ONLY audio output being used ?

Thanks for considering this.

Chris

**Reply by rern** [reply]
Try:
```sh
sed -i '/bluealsa/ {n;n;a\
\tformat         "44100:16:2"
}' /srv/http/bash/mpd-conf.sh
```
Then disconnect > reconnect Bluetooth.

**Reply by ceejayemm** [reply]
@rern 

Yes, that seems to do the trick. My 32 bit floating point encoded albums now seem to play correctly.

Thanks

Chris

**Comment by ceejayemm** [comment]
Further testing has shown that the bit of code provided by rern seem to fix this problem.

Chris

**Comment by ceejayemm** [comment]
@rern 

Forget the above.  I have now found the new 'Sampling 16bit 44.1kHz to receivers' option under the Bluetooth settings on the System page. Turned this on and playing 32 bit floating point albums came back again.  

I have to say this isn't the most obvious thing to do but it seems to work.

Chris

**Comment by ceejayemm** [comment, marked as answer]
Turning on the new 'Sampling 16bit 44.1kHz to receivers' option under the Bluetooth settings on the System page fixed this issue.

Chris
